### ISSUE TITLE
unable to chat in terminal

### ISSUE THREAD

**Issue by Longame208** [elizaOS/eliza#1204]
run pnpm start / pnpm start:client and not able to have chat in terminal with agent was working before now it takes to local host page browser and nothing happens

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by jaycoolh** [comment]
Same error here

**Comment by jaycoolh** [comment]
I am using: v0.1.6-alpha.4

node --version 
v23.3.0

pnpm --version                                                                               
9.15.0

 pnpm start --character="characters/trump.character.json"

**Comment by jaycoolh** [comment]
["◎ Visit the following URL to chat with your agents:"]

 ["◎ http://localhost:5173"]

 ["✓ REST API bound to 0.0.0.0:3000. If running locally, access it at http://localhost:3000."]
 
 
 'http://localhost:5173' is just a dead link

**Comment by jaycoolh** [comment]
@Longame208 

in the package.json there is a script `pnpm start:client`

this spins up the app http://localhost:5173

Need to include this in the quickstart guide https://ai16z.github.io/eliza/docs/quickstart/#next-steps

**Comment by ProphetX10** [comment]
> pnpm start : client

i have tried this but i get this error since I guess it tries to open another rpc server on port 3000?

node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use :::3000

**Comment by jaycoolh** [comment]
For me the webui opened in a different port! maybe you have a PORT=300 in your environment vars?

**Comment by ProphetX10** [comment]
i reinstalled everything in a new folder, and also made sure the error with the canvas install package was corrected and installed, and now everything is fine

**Comment by AIFlowML** [comment]
You want o chat in the terminal or in the Browser ? 
In terminal you not need the Client mate. You just chat in the terminal. 
The client is a separarte web app that let you chat with all the character loaded. 

Here some tips 

#### TLDR (Executive Summary)
- **Issue essence**: Terminal chat functionality fails due to API endpoint misconfiguration.
- **Root cause**: Inaccessible or misconfigured API endpoint `/api/${agentId}/message` in `Chat.tsx`.
- **Solution approach**: Verify server, fix API endpoint, add error handling, and test routing.
- **Impact assessment**: Restores chat functionality and improves user experience.

#### Quick Resolution Path
1. **Verify Server Status**:
   - Ensure the server is running with `pnpm start` or `pnpm start:client`.
2. **Check API Endpoint**:
   - Confirm `/api/${agentId}/message` is accessible and correctly configured.
3. **Add Error Handling**:
   - Update `Chat.tsx` to handle API call failures:
     ```javascript
     const mutation = useMutation({
         mutationFn: async (text: string) => {
             try {
                 const formData = new FormData();
                 formData.append("text", text);
                 formData.append("userId", "user");
                 formData.append("roomId", `default-room-${agentId}`);
                 if (selectedFile) {
                     formData.append("file", selectedFile);
                 }
                 const res = await fetch(`/api/${agentId}/message`, {
                     method: "POST",
                     body: formData,
                 });
                 if (!res.ok) throw new Error("API call failed");
                 return res.json() as Promise<TextResponse[]>;
             } catch (error) {
                 console.error("Error sending message:", error);
                 throw error;
             }
         },
         onSuccess: (data) => {
             setMessages((prev) => [...prev, ...data]);
             setSelectedFile(null);
         },
         onError: (error) => {
             alert("Failed to send message. Please try again.");
         },
     });
     ```
4. **Test the Fix**:
   - Run `pnpm start` and verify chat functionality.

#### Comprehensive Solution
1. **Server Verification**:
   - Ensure the server is running and accessible.
2. **API Endpoint Configuration**:
   - Verify `/api/${agentId}/message` is correctly set up in the backend.
3. **Error Handling**:
   - Implement robust error handling in `Chat.tsx` to provide user feedback.
4. **Routing Configuration**:
   - Review and ensure proper routing for terminal requests.
5. **Testing**:
   - Execute `pnpm start` and test the chat functionality.

**Comment by AIFlowML** [comment]
Closin this issue because the user resolved it.

**Comment by btwiuse** [comment]
Fyi support for terminal chat was removed in https://github.com/elizaOS/eliza/commit/6a7da2445cdb8e8fd1602022145e01ced41b7a9d

If someone wants it back, you can always write a standalone cli based on the deleted part.
